pkg/kn/commands/flags/sink.go

```diff
@@ -107,7 +108,7 @@ func (i *SinkFlags) ResolveSink(knclient clientdynamic.KnDynamicClient, namespac
 		if prefix == "svc" || prefix == "service" {
 			return nil, fmt.Errorf("unsupported sink prefix: '%s', please use prefix 'ksvc' for knative service", prefix)
 		}
-		return nil, fmt.Errorf("unsupported sink prefix: '%s'", prefix)
+		return nil, fmt.Errorf("unsupported sink prefix: '%s', if referring to a knative service in another namespace, 'ksvc:name:namespace' combination must be provided explicitly", prefix)
```

Comment:
Shouldn’t `if` really be `when`?

Reply:
for the special case of knative service, where we may not require a prefix, we cant detect if its ksvc name or a prefix (defined in config)